drivers_linux.go

```diff
@@ -1,25 +1,66 @@
 package libnetwork
 
 import (
+	"strings"
+
 	"github.com/docker/libnetwork/driverapi"
 	"github.com/docker/libnetwork/drivers/bridge"
 	"github.com/docker/libnetwork/drivers/host"
 	"github.com/docker/libnetwork/drivers/null"
 	o "github.com/docker/libnetwork/drivers/overlay"
 	"github.com/docker/libnetwork/drivers/remote"
+	"github.com/docker/libnetwork/netlabel"
 )
 
-func initDrivers(dc driverapi.DriverCallback) error {
-	for _, fn := range [](func(driverapi.DriverCallback) error){
-		bridge.Init,
-		host.Init,
-		null.Init,
-		remote.Init,
-		o.Init,
-	} {
-		if err := fn(dc); err != nil {
+func initDrivers(c *controller) error {
+	initializers := []struct {
+		fn    func(driverapi.DriverCallback, map[string]interface{}) error
+		ntype string
+	}{
+		{bridge.Init, "bridge"},
+		{host.Init, "host"},
+		{null.Init, "null"},
+		{remote.Init, "remote"},
+		{o.Init, "overlay"},
+	}
+
+	for _, i := range initializers {
+		if err := i.fn(c, makeDriverConfig(c, i.ntype)); err != nil {
```

Comment:
Can you please make this change in drivers_windows.go and drivers_freebsd.go as well ?